hw/top_earlgrey/data/top_earlgrey.sv.tpl

```diff
@@ -608,6 +637,10 @@ else:
   );
 
 % endfor
+  // USB assignments
```

Comment:
shouldn't this be done inside the usbdev instead of here in `top_earlgrey.tpl.sv`?

Reply:
For now, I did not want to change the interface of usbdev (other than renaming signals). Mainly to get this into CI quickly without breaking stuff and time consuming debugging.

But I agree, these assignments would better be done inside usbdev. I suggest to do this in a follow-up PR.

Reply:
I guess it's debatable where they belong. I moved them out of the usbdev, because I consider them to be quirks of using FPGA GPIOs for the USB data signals. IMHO they belong in some FPGA-specific part of the IO configuration (e.g. padctl.sv). I have no clue what the reason is for having two pull-up signals...

Reply:
Ah I did not realize that. Well, the reason for having two pull-up signals and separate enables for `usb_dp_o` and `usb_dn_o` is our register tool: For every output, the tool generates also an output enable.

Reply:
I would like to suggest that we should have shim wrapper then. Not saying fixed in this PR but as a TODO.

Reply:
I have created a tracking issue for USB and made a note there: #1403